lib/game.rb

```diff
@@ -0,0 +1,176 @@
+
+    rand_sub_coords = all_sub_placements.sample
+    until @computer_player.board.valid_placement?(@computer_player.submarine, rand_sub_coords) do
+      rand_sub_coords = all_sub_placements.sample
+    end
+    @computer_player.board.place(@computer_player.submarine, rand_sub_coords)
+  end
+
+  def player_ship_placement_message
+    puts "I have laid out my ships on the grid."
+    puts "You now need to lay out your two ships."
+    puts "The Cruiser is three units long and the Submarine is two units long."
+    puts "#{@human_player.board.render}"
+    puts "Enter the squares for the Cruiser (3 spaces):"
+    print ">"
+  end
+
+  def player_ship_placement_user_input
+    player_ship_placement_message
+    cruiser_input = gets.chomp.split(" ").to_a
+    until all_cruiser_placements.include?(cruiser_input) && @human_player.board.valid_placement?(@human_player.cruiser, cruiser_input) do
+      puts "Those are invalid coordinates. Plese try again:"
+      # puts "Ships should be placed on linear, consecutive, cells from left to right or top to bottom."
+      cruiser_input = gets.chomp.split(" ").to_a
+    end
+    @human_player.board.place(@human_player.cruiser, cruiser_input)
+    puts "#{@human_player.board.render(true)}"
+
+    puts "Enter the squares for the Submarine (2 spaces)"
+    print ">"
+    sub_input = gets.chomp.split(" ").to_a
+    until all_sub_placements.include?(sub_input) && @human_player.board.valid_placement?(@human_player.submarine, sub_input) do
+      puts "Those are invalid coordinates. Plese try again:"
+      # puts "Ships should be placed on linear, consecutive, cells from left to right or top to bottom."
+      sub_input = gets.chomp.split(" ").to_a
+    end
+    @human_player.board.place(@human_player.submarine, sub_input)
+    puts "#{@human_player.board.render(true)}"
+  end
+
+  # def valid_coordinate_format?(input)
+  #    #iterate through the input to see if its valid
+  #    input.all? do |coordinate|
+  #      coordinate.match?(/\w{2}/)
+  #    end
+  # end
+
+  def turn
+    # dorion put the top part in a method
+    until (@human_player.cruiser.sunk? && @human_player.submarine.sunk?) || (@computer_player.cruiser.sunk? && @computer_player.submarine.sunk?) do
+      @game_turn.board_display
+      @game_turn.player_shot
+      @game_turn.computer_shot
+    end
+    if (@human_player.cruiser.sunk? && @human_player.submarine.sunk?) && (@computer_player.cruiser.sunk? && @computer_player.submarine.sunk?)
+      @game_turn.board_display
+      puts "Its a draw!"
+    elsif @human_player.cruiser.sunk? && @human_player.submarine.sunk?
+      @game_turn.board_display
+      puts "I won"
+    elsif @computer_player.cruiser.sunk? && @computer_player.submarine.sunk?
+      @game_turn.board_display
+      puts "You won"
+    end
+    setup
+    start
+  end
+
+# Initializes an array with all possible placement array combinateions for the cruiser
+  def all_cruiser_placements
```

Comment:
creating an array of all the possible placements for both ships really came in handy and made the placement methods a lot easier to work with